index.bs

```diff
@@ -692,7 +692,7 @@ partial interface MLModelBuilder {
     if (!hiddenState) {
       const desc = { type: 'float32', dimensions: [numDirections, 1, hiddenSize] };
       const totalSize = numDirections * hiddenSize;
-      hiddenState = builder.constant(desc, new Float32Array(totalSize).fill(0));
+      hiddenState = builder.constant(desc, new Float32Array(totalSize).fill(0).buffer);
```

Comment:
According to `dictionary MLBuffer`, it would require the `resource` field, e.g.
```js
hiddenState = builder.constant(desc, {resource: new Float32Array(totalSize).fill(0).buffer});
```

Or should we define the `MLBuffer` as a union of buffer type and buffer range?, e.g.
```js
typedef (ArrayBuffer or WebGLBuffer or GPUBuffer) MLBufferType;

dictionary MLBufferRange {
  required MLBufferType buffer;
  unsigned long long offset = 0;
  unsigned long long size;
};

typedef (MLBufferRange or MLBufferType) MLBuffer;
```

With that, the original code works.
```js
hiddenState = builder.constant(desc, new Float32Array(totalSize).fill(0).buffer);
```

WDYT?

Reply:
@huningxin The `MLBufferRange` is really a buffer view while `MLBufferType` is a buffer resource type. Calling either a buffer resource or a buffer view a `MLBuffer` could be confusing.

I propose in my latest commit a notion of `MLBufferView` which could either be a standard array buffer view or a buffer resource view.

``` javascript
dictionary MLBufferResourceView {
  required (WebGLBuffer or GPUBuffer) resource;
  unsigned long long offset = 0;
  unsigned long long size;
};
 
typedef (ArrayBufferView or MLBufferResourceView) MLBufferView;

interface MLModelBuilder {
  ...
  MLOperand constant(MLOperandDescriptor desc, MLBufferView bufferView);
};
```

This way this code can just look like this:

``` javascript
      hiddenState = builder.constant(desc, new Float32Array(totalSize).fill(0));
```
It then follows that for the input/output, the types are the following: Note that the field's name is now simply `data` as opposed to `resource` to avoid the confusion with the GPU resource types.

``` javascript
dictionary MLInput {
  required (MLBufferView or WebGLTexture or GPUTexture) data;
  sequence<long> dimensions;
};

dictionary MLOutput {
  (MLBufferView or WebGLTexture or GPUTexture) data;
  sequence<long> dimensions;
};

...
// 3. Bind inputs to the model and execute for the result.
const bufferA = new Float32Array(4).fill(1.0);
const bufferB = new Float32Array(4).fill(0.8);
const inputs = {'A': {data: bufferA}, 'B': {data: bufferB}};
const outputs = await compilation.compute(inputs);
...
```
If this all looks good, I'll prepare the last commit that renames everything to:

> MLModelBuilder -> MLGraphBuilder
MLModelBuilder.compile -> MLGraphBuilder.build
MLCompilation -> MLGraph

Reply:
Thanks @wchao1115 . Your proposal in the latest commit https://github.com/webmachinelearning/webnn/pull/149/commits/be764ba4c32c40f77ec7c45cb70a12137e95c5c2 looks good to me.